public/src/components/Home/SidebarComponent.jsx

```diff
@@ -0,0 +1,144 @@
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class SidebarComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchFormValue: '',
+            archiveFormValue: '',
+        };
+        this.getSubredditsList = this.getSubredditsList.bind(this);
+        this.onChangeSearchForm = this.onChangeSearchForm.bind(this);
+        this.onChangeArchiveForm = this.onChangeArchiveForm.bind(this);
+        this.getArchiveFormClass = this.getArchiveFormClass.bind(this);
+        this.onKeyPressSearchForm = this.onKeyPressSearchForm.bind(this);
+        this.setSearchValue = this.setSearchValue.bind(this);
+    }
+
+    onKeyPressSearchForm(event) {
+        const { searchFormValue } = this.state;
+        const { storeSubredditToArchive } = this.props;
+        if (searchFormValue && event.key === 'Enter') {
+            storeSubredditToArchive(searchFormValue);
+            this.setState({
+                searchFormValue: '',
+            });
+        }
+    }
+
+    onChangeSearchForm(event) {
+        this.setState({
+            searchFormValue: event.target.value,
+        });
+    }
+
+    onChangeArchiveForm(event) {
+        this.setState({
+            archiveFormValue: event.target.value,
+        });
+    }
+
+    setSearchValue(value) {
+        this.setState({
+            searchFormValue: value,
+        });
+    }
+
+    getArchiveFormClass() {
+        const { subreddits } = this.props;
+        const minimumArchivedLength = 5;
+        const archivedSubreddits = [...subreddits]
+            .filter(subreddit => subreddit.isArchived);
+        return archivedSubreddits.length > minimumArchivedLength
+            ? 'home-sidebar__archive-form'
+            : 'home-sidebar__archive-form disable';
+    }
+
+    getSubredditsList() {
+        const { subreddits } = this.props;
+        const { searchFormValue } = this.state;
+        const subredditList = [...subreddits].map((subreddit, key) => {
+            const formValue = searchFormValue
+                .replace(/\\/g, "\\\\")
+                .toLowerCase()
+                .trim();
+            const subredditTitle = subreddit.display_name
+                .toLowerCase()
+                .trim()
+                .slice(0, subreddit.display_name.length - 1);
+
+            if (formValue && subredditTitle.search(formValue) > -1 && !subreddit.isArchived) {
+                return (
+                    <div key={key}
+                         tabIndex={0}
+                         role="button"
+                         onClick={() => this.setSearchValue(subreddit.display_name)}
+                    >
+                        {subreddit.display_name}
+                    </div>
+                );
+            }
+            return null;
+        });
+
+        if (subredditList.some(subreddit => subreddit)) {
+            const trueValue = true;
+            const falseValue = true;
```

Comment:
Looks interesting =)

Reply:
) 
it's eslint stuff. airbnb mode prohibits to use:
stopScrollPropagation={true}
horizontal={false}

Reply:
fixed https://github.com/YuriyBC/Reddit-archive/commit/2471bcb527f4e614fa1989bcf345be29b840c8a5